ddtrace/llmobs/_integrations/utils.py

```diff
@@ -397,6 +397,209 @@ def openai_set_meta_tags_from_chat(span: Span, kwargs: Dict[str, Any], messages:
     span._set_ctx_item(OUTPUT_MESSAGES, output_messages)
 
 
+def openai_set_meta_tags_from_response(span: Span, kwargs: Dict[str, Any], response: Optional[Any]) -> None:
+    """Extract input/output tags from response and set them as temporary "_ml_obs.meta.*" tags."""
+    input_data = kwargs.get("input", [])
+    input_messages = []
+    
+    if isinstance(input_data, str):
+        input_messages.append({"content": input_data, "role": "user"})
+    elif isinstance(input_data, list):
+        for m in input_data:
+            role = m.get("role", "user")
+            content = m.get("content", "")
+            if isinstance(content, list):
+                content_text = ""
+                for content_item in content:
+                    if isinstance(content_item, dict) and "text" in content_item:
+                        content_text += content_item.get("text", "")
+                    elif isinstance(content_item, str):
+                        content_text += content_item
+                    else:
+                        content_text += "[Unsupported content item type of {}]".format(type(content_item))
+            elif isinstance(content, str):
+                content_text = content
+            else:
+                content_text = "[Unsupported content type of {}]".format(type(content))
+            input_messages.append({"content": content_text, "role": role})
+    else:
+        input_messages.append({"content": "[Unsupported content type of {}]".format(type(input_data))})
+
+    # Add instructions to as an input system messages
+    if "instructions" in kwargs:
+        input_messages.insert(0, {"content": kwargs["instructions"], "role": "system"})
+    parameters = {
+        k: v for k, v in kwargs.items() if k not in ("model", "input", "api_key", "user_api_key", "user_api_key_hash")
+    }
+    span._set_ctx_items({INPUT_MESSAGES: input_messages, METADATA: parameters})
+    if span.error or not response:
+        span._set_ctx_item(OUTPUT_MESSAGES, [{"content": ""}])
+        return
+
+    # For streamed responses
+    if kwargs.get("stream"):
+        # To discuss: should I remove usage from streamed output message?
+        for item in response:
+            if "usage" in item:
+                item.pop("usage")
+        span._set_ctx_item(OUTPUT_MESSAGES, response)
+    # For non-streamed responses
+    # else:
+    output_data = _get_attr(response, "output", [])
+    output_messages = []
+    for output in output_data:
+        # Message or reasoning
+        if output.type == "message" or output.type == "reasoning":
```

Comment:
I feel like there is a ton of duplicate code here and in `openai_construct_response_from_streamed_chunks`. Can we ideally try combining this? If it gets too complicated, then no worries.